op_crates/crypto/lib.rs

```diff
@@ -41,3 +48,43 @@ pub fn op_get_random_values(
 
   Ok(json!({}))
 }
+
+lazy_static! {
+  static ref DIGEST_ALGORITHMS_BY_NAME: HashMap<&'static str, &'static ring::digest::Algorithm> = {
+    let mut hm = HashMap::new();
+    hm.insert("SHA-1", &ring::digest::SHA1_FOR_LEGACY_USE_ONLY);
+    hm.insert("SHA-256", &ring::digest::SHA256);
+    hm.insert("SHA-384", &ring::digest::SHA384);
+    hm.insert("SHA-512", &ring::digest::SHA512);
+    hm
+  };
+}
+
+pub async fn op_subtle_digest(
+  _state: Rc<RefCell<OpState>>,
+  args: Value,
+  zero_copy: BufVec,
+) -> Result<Value, AnyError> {
+  let mut zc = zero_copy;
+
+  // FIXME: This chain of `unwrap` is meh.
+  let alg_name = args
+    .as_object()
+    .unwrap()
+    .get("name")
+    .unwrap()
+    .as_str()
+    .unwrap();
+  let digest_algorithm = DIGEST_ALGORITHMS_BY_NAME.get(alg_name).unwrap();
+  // TODO: This should run in another thread
+  // Otherwise, even though the API is async, it is blocking the main thread
+  // See https://stackoverflow.com/questions/61292425/how-to-run-an-asynchronous-task-from-a-non-main-thread-in-tokio
```

Comment:
From what I understand, `jsonOpAsync()` doesn't necessarily mean that the code runs asynchronously. It just means that it returns a `Promise` (a `Future` on Rust side).
So, to make it truly asynchronous, we'd have to run the call to ring's `digest()` in another thread. I have no idea how to do this. Is there an example of this already in the codebase?

Reply:
There's no point in making this op async if the Rust API is sync; just use sync op and if you need to return promise do it manually in JS

Reply:
If you have some synchronous CPU intensive operation with an async API in JS, you can run it in the tokio blocking thread pool (tokio::task::spawn_blocking).

Reply:
Good point 👍

Reply:
@lucacasonato Thank you very much, it's exactly what I was looking for.